src/engine/strat_engine/filesystem.rs

```diff
@@ -109,6 +153,30 @@ impl Recordable<FilesystemSave> for StratFilesystem {
     }
 }
 
+/// Return total blocks, free blocks, blocksize for a filesystem
+pub fn usage(mount_point: &PathBuf) -> EngineResult<(u64, u64, u64)> {
+    let mut stat = vfs::Statvfs::default();
+    try!(statvfs(mount_point, &mut stat));
+    Ok((stat.f_blocks, stat.f_bfree, stat.f_bsize))
```

Comment:
Easier on caller if we multiply `blocks` and `bfree` by `bsize` in here? There is some filesystem-block size versus thinpool DataBlock shenanigans going on up in `check`, which might be made clearer if the fact that statvfs returns things in "filesystem blocks" was not exposed to callers. Maybe return `(Bytes, Bytes)`?

Consider renaming to `fs_usage` or something else a little more specific than `usage`?

Consider returning total/used instead of total/free?